src/diffusers/utils/callbacks.py

```diff
@@ -0,0 +1,116 @@
+from functools import partial
+from typing import Dict, List
+
+
+class PipelineCallback:
+    def __init__(self, **kwargs):
+        self.func = partial(self.callback_fn, **kwargs)
```

Comment:
@yiyixuxu IMO it's better to use the callback args as instance variables e.g.: `self.step_ratio = step_ratio` than `**kwargs` but I'll keep this in case there's a use case in the future where is needed WDYT?

Reply:
happy to see a proposal!

Reply:
I did it in the callback functions. The new callback functions aren't using the `**kwargs` anymore.

Reply:
ahhh I see

if `step_ratio` is something we want to give to all our callbacks (I think we probably should), maybe we can make class PipelineCallback a subclass of `ConfigMixin`, i.e. 

```python
class PipelineCallback(ConfigMixin):
    @register_to_config
    def __init__(self, step_ratio=1.0):
```

Reply:
oh I like that idea, maybe in that case we can use `start_step` and `end_step` instead and cover all the future use cases.

Reply:
we probably will need them but let's add them later when we need them:)
for now, I think step_ratio is sufficient, maybe name it `cutoff_step_ratio`? and do you think it makes sense to also add a `cutoff_step_index`?

Reply:
I like the idea of `cutoff_step_index`  but we need to set a default and I don't think it's a good idea to set it to 0 but we don't know the exact value of the `step_index` here, so set it to 1000?